src/main/java/ru/mercuriev/game/of/life/graph/GraphEngine.java

```diff
@@ -0,0 +1,33 @@
+package ru.mercuriev.game.of.life.graph;
+
+import ru.mercuriev.game.of.life.graph.cell.Cell;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Engine interface implementation
+ * Uses Cell uder the hood to calculate next generation
```

Comment:
typo - "under"
I think you may mention about links to other cells and "no-array" implementation as a goal of this Engine.
"Cell under the hood@ is not clear. What cell? What are cell's abilities? So on.